pkg/signatures/cosign_sig_verifier.go

```diff
@@ -160,14 +161,30 @@ func (c *cosignSignatureVerifier) VerifySignature(ctx context.Context,
 		allVerifyErrs = multierror.Append(allVerifyErrs, err)
 	}
 
+	// Find the union of all image references from verified signatures. The resulting status
+	// is verified if at least one verification was successful.
+	//
+	// verifier_1(sig_1) OR ... OR verifier_1(sig_N)
+	// OR
+	// ...
+	// OR
+	// verifier_N(sig_1) OR ... OR verifier_N(sig_N)
+	verifiedImageReferences := set.NewStringSet()
 	for _, opts := range c.verifierOpts {
-		verifiedImageReferences, err := verifyImageSignatures(ctx, sigs, hash, image, opts)
-		if err == nil && len(verifiedImageReferences) != 0 {
-			return storage.ImageSignatureVerificationResult_VERIFIED, verifiedImageReferences, nil
+		verifierReferenceSet, err := verifyImageSignatures(ctx, sigs, hash, image, opts)
```

Comment:
nit, question: wouldn't extracting the loop out of `verifyImageSignatures` make the code a bit simpler?

Reply:
yes it would, good point.